Facebook.AccountKit.iOS/source/Facebook.AccountKit/ApiDefinition.cs

```diff
@@ -0,0 +1,604 @@
+        void SetActionController(IActionController actionController);
+
+        // @optional -(void)setError:(NSError * _Nonnull)error;
+        [Export("setError:")]
+        void SetError(NSError error);
+
+        // @optional -(AKFTextPosition)textPositionForState:(AKFLoginFlowState)state;
+        [Export("textPositionForState:")]
+        TextPosition GetTextPosition(LoginFlowState state);
+
+        // @optional -(Theme * _Nullable)theme;
+        [NullAllowed, Export("theme")]
+        //[Verify(MethodToProperty)]
+        Theme GetTheme();
+    }
+
+    interface IUIManager { }
+
+    // @protocol AKFAdvancedUIManager <AKFUIManager>
+    [Protocol (Name = "AKFAdvancedUIManager")]
+    interface AdvancedUIManager : UIManager
+    {
+    }
+
+    interface IAdvancedUIManager { }
+
+    // @protocol AKFAdvancedUIActionController <AKFActionController>
+    [Protocol (Name = "AKFAdvancedUIActionController")]
+    interface AdvancedUIActionController : ActionController
+    {
+    }
+
+    interface IAdvancedUIActionController { }
+
+    // @protocol AKFUIManaging <NSObject>
+    [Protocol (Name = "AKFUIManaging")]
+    interface UIManaging
+    {
+        // @required @property (nonatomic, strong) id<AKFUIManager> uiManager;
+        [Abstract]
+        [Export("uiManager", ArgumentSemantic.Strong)]
+        IUIManager UiManager { get; set; }
+
+        // @required -(void)setAdvancedUIManager:(id<AKFAdvancedUIManager>)uiManager;
+        [Abstract]
+        [Export("setAdvancedUIManager:")]
+        void SetAdvancedUIManager(IAdvancedUIManager uiManager);
+
+        // @required -(void)setTheme:(Theme *)theme;
+        [Abstract]
+        [Export("setTheme:")]
+        void SetTheme(Theme theme);
+    }
+
+    interface IUIManaging { }
+
+    // @protocol AKFViewController <AKFUIManaging, AKFConfiguring>
+    [Protocol (Name = "AKFViewController")]
+    interface ViewController : UIManaging, Configuring
+    {
+        // @required @property (nonatomic, weak) id<AKFViewControllerDelegate> delegate;
+        [Abstract]
+        [Export("delegate", ArgumentSemantic.Weak)]
+        IViewControllerDelegate Delegate { get; set; }
+
+        // @required @property (readonly, assign, nonatomic) AKFLoginType loginType;
+        [Abstract]
+        [Export("loginType", ArgumentSemantic.Assign)]
+        LoginType LoginType { get; }
+    }
+
+    interface IViewController { }
+
+    // @protocol AKFViewControllerDelegate <NSObject>
+    [Protocol, Model]
+    [BaseType(typeof(NSObject), Name = "AKFViewControllerDelegate")]
+    interface ViewControllerDelegate
+    {
+        // @optional -(void)viewController:(UIViewController<AKFViewController> *)viewController didCompleteLoginWithAuthorizationCode:(NSString *)code state:(NSString *)state;
+        [Export("viewController:didCompleteLoginWithAuthorizationCode:state:")]
+        [EventArgs("ViewControllerDidCompleteLoginWithCode")]
+        [EventName("LoginCompleted")]
```

Comment:
EventArgs - Not 100% on naming convention desired, I did abbreviate it a little so the generated class name would not be so long.

Reply:
Here we cannot apply events, due the IViewControllerDelegate's `Delegate` property is inside of the `ViewController` protocol. The `Delegate` property must live inside of a class so events can be generated.